From d0a8045c9edc28084a5fc504865c0e47b95983c9 Mon Sep 17 00:00:00 2001
Message-Id: <patch-3.6-d0a8045c9ed-20210908T151929Z-avarab@gmail.com>
In-Reply-To: <patch-1.6-b5c79d87847-20210908T151929Z-avarab@gmail.com>
References: <patch-1.6-b5c79d87847-20210908T151929Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 7 Jun 2021 17:19:40 +0200
Subject: [PATCH 3/6] help tests: add test for --config output
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Cc: Junio C Hamano <gitster@pobox.com>,
    Philip Oakley <philipoakley@iee.email>,
    Nguyễn Thái Ngọc Duy <pclouds@gmail.com>,
    SZEDER Gábor <szeder.dev@gmail.com>

Add a missing test for checking what the --config output added in
ac68a93fd2 (help: add --config to list all available config,
2018-05-26) looks like. We should not be emitting anything except
config variables and the brief usage information at the end here.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t0012-help.sh | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/t/t0012-help.sh b/t/t0012-help.sh
index 6e01da614f0..94d1f481c8b 100755
--- a/t/t0012-help.sh
+++ b/t/t0012-help.sh
@@ -78,6 +78,19 @@ test_expect_success 'git help -g' '
 	test_i18ngrep "^   tutorial   " help.output
 '
 
+test_expect_success 'git help -c' '
+	git help -c >help.output &&
+	cat >expect <<-\EOF &&
+
+	'"'"'git help config'"'"' for more information
+	EOF
+	grep -v -E \
+		-e "^[^.]+\.[^.]+$" \
+		-e "^[^.]+\.[^.]+\.[^.]+$" \
+		help.output >actual &&
+	test_cmp expect actual
+'
+
 test_expect_success 'generate builtin list' '
 	git --list-cmds=builtins >builtins
 '
-- 
2.33.0.825.g2bf60429931

